kedro-org/kedro-viz
# Visualize Dataset statistics in metadata panel

## Description

Resolves #662 

## Development notes

![Dataset_stats_compressed](https://github.com/kedro-org/kedro-viz/assets/87735534/cfe261e2-f975-46ab-a70a-763d1f3dcc5d)

**Screenshots:**

**All stats -**

<img width="2048" alt="Screenshot 2023-08-02 at 8 18 07 AM" src="https://github.com/kedro-org/kedro-viz/assets/87735534/0e2228b3-ab12-4d3f-8a2c-8576dd7c388c">

**File size only -**

<img width="2048" alt="Screenshot 2023-08-02 at 8 22 02 AM" src="https://github.com/kedro-org/kedro-viz/assets/87735534/56a4dfe7-c578-4201-b755-2b63563dc74a">

**File size not configured due to no file path -**

<img width="2045" alt="Screenshot 2023-08-02 at 8 23 24 AM" src="https://github.com/kedro-org/kedro-viz/assets/87735534/ca4fc159-c5f4-4ba5-a2a8-34509a436221">

**Overflow Dataset stats -**

<img width="2045" alt="Screenshot 2023-08-02 at 8 25 01 AM" src="https://github.com/kedro-org/kedro-viz/assets/87735534/8b4ac175-425d-4b94-a3e9-c336bb927a26">

## QA notes

**Steps to QA :**
1. Checkout feature/viz-size-datasets
2. Pip install -e package
3. cd demo-project
4. Execute kedro run
5. A file stats.json will be created in the demo-project along with changes in the data folder due to new kedro run
6. cd .. (go into kedro-viz folder)
7. Start the backend server executing make run
8. Start frontend by executing npm start
9. Click on datasets (covered only instances of pd.Dataframe). You should see the file statistics (rows, columns and file size). If file size is not available, you will see it as NA

**Note:**

1. There needs to be further discussion on where to extract file size as it is not available at the hook level (or I do not know how to get file path at hook level). As of now, I am using fsspec inside the flowchart model to get the file size. 
2. Transcoded data name is stored as ingestion.int_typed_shuttles@pandas2

## Checklist



<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1472"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
